packages/babel-plugin-transform-inline-environment-variables/src/index.js

```diff
@@ -1,11 +1,13 @@
 export default function ({ types: t }) {
   return {
     visitor: {
-      MemberExpression(path) {
+      MemberExpression(path, state) {
         if (path.get("object").matchesPattern("process.env")) {
```

Comment:
Agreed. It still replaces keys matching `process.env`. The issue I had is that, when running babel for multiple target environments, it's more than a little inconvenient to fork/spawn separate processes for the purpose of differentiating the env vars of each.

Since it's even sketchier to overwrite keys in `process.env` within the running script, it seemed like a good compromise. You're still acting on the target's assumption for what `process.env` is going to be -- it just doesn't matter that it's actually running in a discrete process.